osmoutils/binary_search_test.go

```diff
@@ -74,12 +89,19 @@ func TestBinarySearchNonlinear(t *testing.T) {
 }
 
 func TestBinarySearchNonlinearNonzero(t *testing.T) {
```

Comment:
Why is this called "NonLinear" when we have both linear and non-linear functions?

Reply:
Ah this is called nonlinear because it tests nonzero tolerance with a nonlinear function, I can just change it to TestBinarySearchNonzero if that's better?